tests/manage/pv_services/test_resource_deletion_during_pod_pvc_deletion.py

```diff
@@ -0,0 +1,338 @@
+        for pvc_obj in self.pvc_objs:
+            pvc_uuid_map[pvc_obj.name] = pvc_obj.image_uuid
+
+        # Start deleting PVCs
+        pvc_bulk_delete = executor.submit(delete_pvcs, self.pvc_objs)
+
+        if operation_to_disrupt == 'delete_pvcs':
+            ret = self.verify_resource_deletion(
+                get_all_pvcs, initial_num_of_pvc
+            )
+            assert ret, "Wait timeout: PVCs are not being deleted."
+            logging.info(
+                f"PVCs deletion has started."
+            )
+            disruption.delete_resource()
+
+        pvcs_deleted = pvc_bulk_delete.result()
+
+        assert pvcs_deleted, "Deletion of PVCs failed."
+
+        # Verify PVCs are deleted
+        for pvc_obj in self.pvc_objs:
+            assert pvc_obj.ocp.wait_for_delete(pvc_obj.name), (
+                f"PVC {pvc_obj.name} is not deleted"
+            )
+        logging.info("Verified: PVCs are deleted.")
+        self.pvc_objs.clear()
+
+        # Verify PVs are deleted
+        for pv_obj in pv_objs:
+            assert pv_obj.ocp.wait_for_delete(pv_obj.name), (
+                f"PV {pv_obj.name} is not deleted"
+            )
+        logging.info("Verified: PVs are deleted.")
+
+        # Verify PV using ceph toolbox. Image/Subvolume should be deleted.
+        for pvc_name, uuid in pvc_uuid_map.items():
+            if self.interface == constants.CEPHBLOCKPOOL:
+                ret = verify_volume_deleted_in_backend(
+                    interface=self.interface, image_uuid=uuid,
+                    pool_name=self.cbp_obj.name
+                )
+            if self.interface == constants.CEPHFILESYSTEM:
+                ret = verify_volume_deleted_in_backend(
+                    interface=self.interface, image_uuid=uuid
+                )
+            assert ret, (
+                f"Volume associated with PVC {pvc_name} still exists "
+                f"in backend"
+            )
+
+        # Verify number of pods of type 'resource_to_delete'
+        final_num_resource_to_delete = len(pod_functions[resource_to_delete]())
+        assert final_num_resource_to_delete == num_of_resource_to_delete, (
+            f"Total number of {resource_to_delete} pods is not matching with "
+            f"initial value. Total number of pods before deleting a pod: "
+            f"{num_of_resource_to_delete}. Total number of pods present now: "
+            f"{final_num_resource_to_delete}"
+        )
+
+        # Check ceph status
+        ceph_health_check(namespace=config.ENV_DATA['cluster_namespace'])
+        log.info("Ceph cluster health is OK")
+
+
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_ceph_block_pool.__name__,
+    create_rbd_storageclass.__name__,
+    create_project.__name__,
+    create_pvcs.__name__,
+    create_pods.__name__
+)
+@tier4
+class TestDeleteResourceRBD(DisruptionBase):
+    """
+    RBD related disruption tests class
+    """
+    interface = constants.CEPHBLOCKPOOL
```

Comment:
are you going to add also cephfilesystem?

Reply:
Implementation has changed. Now using only one test method.